File: apps/api/src/schemas/live.py
from pydantic import BaseModel

from src.schemas.conjunction import ConjunctionEventSummary
from src.schemas.object import FeedStatus, TrackedObjectSummary


class LiveSnapshot(BaseModel):
    epoch: str
    objects: list[TrackedObjectSummary]
    conjunctions: list[ConjunctionEventSummary]
    feeds: list[FeedStatus]

File: packages/api-client/src/endpoints/live.ts
import type { ConjunctionEventSummary, FeedStatus, TrackedObjectSummary } from "@sdmps/domain";
import { defaultApiClient } from "../client";

export type LiveSnapshot = {
  epoch: string;
  objects: TrackedObjectSummary[];
  conjunctions: ConjunctionEventSummary[];
  feeds: FeedStatus[];
};

export async function getLiveSnapshot() {
  return defaultApiClient.get<LiveSnapshot>("/v1/live/snapshot");
}
